### ISSUE TITLE
request: make TreeIndex::range accept Borrowed keys instead of K

### ISSUE THREAD

**Issue by qthree** [wvwwvwwv/scalable-concurrent-containers#158]
BTreeMap in std allows to use `Borrow`ed keys
```rust
pub fn range<T, R>(&self, range: R) -> Range<'_, K, V> ⓘ
    where
        T: Ord + ?Sized,
        K: Borrow<T> + Ord,
        R: RangeBounds<T>,
```
TreeIndex forces usage of K
Together with inability to peek K (#157) it's almost impossible to use TreeIndex::range without reallocating keys all the time.

**Comment by wvwwvwwv** [comment]
Good idea, I’ll work on this next week.

**Comment by qthree** [comment]
> Good idea, I’ll work on this next week.

Implemented in #159

**Comment by changgyoopark-db** [comment]
=> SCC 2.1.18
